freebsd/Platform.c

```diff
@@ -230,31 +250,34 @@ void Platform_setMemoryValues(Meter* this) {
    const FreeBSDMachine* fhost = (const FreeBSDMachine*) host;
 
    this->total = host->totalMem;
-   this->values[MEMORY_METER_USED] = host->usedMem;
-   this->values[MEMORY_METER_SHARED] = host->sharedMem;
-   // this->values[MEMORY_METER_COMPRESSED] = "compressed memory, like zswap on linux"
-   this->values[MEMORY_METER_BUFFERS] = host->buffersMem;
-   this->values[MEMORY_METER_CACHE] = host->cachedMem;
-   // this->values[MEMORY_METER_AVAILABLE] = "available memory"
+   if (host->settings->showCachedMemory) {
+      this->values[MEMORY_CLASS_WIRED]    = fhost->wiredMem;
+      this->values[MEMORY_CLASS_BUFFERS]  = fhost->buffersMem;
+   }
+   else { // if showCachedMemory is disabled, merge buffers into the wired pages
+      this->values[MEMORY_CLASS_WIRED]    = fhost->wiredMem + fhost->buffersMem;
+      this->values[MEMORY_CLASS_BUFFERS]  = 0;
+   }
+   this->values[MEMORY_CLASS_ACTIVE]   = fhost->activeMem;
+   this->values[MEMORY_CLASS_LAUNDRY]  = fhost->laundryMem;
+   this->values[MEMORY_CLASS_INACTIVE] = fhost->inactiveMem;
 
    if (fhost->zfs.enabled) {
       // ZFS does not shrink below the value of zfs_arc_min.
       unsigned long long int shrinkableSize = 0;
       if (fhost->zfs.size > fhost->zfs.min)
          shrinkableSize = fhost->zfs.size - fhost->zfs.min;
-      this->values[MEMORY_METER_USED] -= shrinkableSize;
-      this->values[MEMORY_METER_CACHE] += shrinkableSize;
-      // this->values[MEMORY_METER_AVAILABLE] += shrinkableSize;
+      this->values[MEMORY_CLASS_ARC] = shrinkableSize;
    }
+   else
+      this->values[MEMORY_CLASS_ARC] = 0;
```

Comment:
```suggestion
   } else {
      this->values[MEMORY_CLASS_ARC] = 0;
   }
```

For if statements prefer either both branches with braces or none.